pages/blog/posts/understanding-lexical-dynamic-scopes.md

```diff
@@ -0,0 +1,244 @@
+`/properties/baz` is part of the root schema resource:
+
+![A JSON Schema that consists of 3 schema resources](/img/posts/2024/understanding-lexical-dynamic-scopes/schema-resources.webp)
+
+Going back to scopes, <i>an important rule to remember is that the lexical scope
+of a schema does not extend past its schema resource</i>.
+
+As an example of the relationship between schema resources and lexical scopes,
+consider the `$anchor` keyword that defines a [location-independent
+identifier](https://json-schema.org/draft/2020-12/json-schema-core#section-8.2.2)
+for a schema. While declaring the same anchor identifier more than once in the
+same schema resource is an error, it is possible to declare the same anchor
+identifier on different schema resources within the same schema. This is
+possible because <i>schema resources do not share their lexical scopes</i>:
+
+![Defining multiple anchors with the same name on different schema resources](/img/posts/2024/understanding-lexical-dynamic-scopes/static-anchors.webp)
+
+### Changing Scopes With References
+
+Now that we understand lexical scopes and their boundaries within schema
+resources, let's consider the effect that following references has on them.
+<i>When evaluation encounters a reference keyword, it abandons the lexical scope
+of the reference schema and enters the lexical scope of the destination
+schema</i>. In other words, the evaluation process needs to re-calculate the
+lexical scope of the schema it ends up at after following the reference.
+
+Consider the following example. In the left, the lexical scope of a subschema
+that defines a static reference. In the right, the lexical scope of the
+destination subschema after following such reference. Note that the destination
+is part of a different schema resource, under a different meta-schema. The
+change of lexical scope let's the destination schema be governed, as expected,
+by its own meta-schema instead of by the meta-schema of the root schema
+resource:
+
+![Lexical scope before and after following a static reference](/img/posts/2024/understanding-lexical-dynamic-scopes/ref.webp)
+
+As an interesting scenario, note that the root schema resource is not
+necessarily in the lexical scope of the first subschema that the evaluation
+process will visit. Consider a remote reference that points to a nested schema
+resource. In this case, the evaluation process jumps to the lexical scope of
+the destination schema resource, which may not include the root schema
+resource. For example:
+
+![Following a remote reference to a nested schema resource](/img/posts/2024/understanding-lexical-dynamic-scopes/ref-anchor.webp)
+
+Dynamic Scope
+-------------
+
+To recap, the lexical scope of a schema consists of the schema itself and its
+corresponding parent schemas within the same schema resource. In comparison,
+<i>the dynamic scope of a schema consists of the subschemas that were evaluated
```

Comment:
Minor, but might be more effective to emphasize *less* in this sentence, so the important bits stand out. E.g.:

```suggestion
the *dynamic* scope of a schema consists of the subschemas that were *evaluated so far*
```

Reply:
Makes sense!